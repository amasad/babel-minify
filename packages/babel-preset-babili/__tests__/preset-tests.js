jest.autoMockOff();

const transformer = require("../../../utils/test").transformer;
const transform = transformer(void 0, {
  presets: [require("../src")]
});

describe("preset", () => {
  // https://github.com/babel/babili/issues/122
  it ("should fix issue#122", () => {
    const source = `
      function foo() {
        var a, b, c;
        if (a) {
          if (b) {
            if (c) {}
          }
        } else {
          if (b) {
          } else {
            if (c) {}
          }
        }
      }
    `;

    expect(transform(source)).toMatchSnapshot();
  });
});
