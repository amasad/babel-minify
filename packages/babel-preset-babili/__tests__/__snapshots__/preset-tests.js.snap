exports[`preset should fix issue#122 1`] = `
"
function foo() {
  var d, e, f;
  d ? e && f : !b && f;
}"
`;
